runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ExecutionState.java

```diff
@@ -35,8 +42,13 @@ public abstract class ExecutionState {
 
   public abstract ExecutionStateTracker executionStateTracker();
 
-  public static ExecutionState.Builder builder() {
-    return new AutoValue_ExecutionState.Builder();
+  public final void close() {
```

Comment:
should we have this state be autoclosable?

Reply:
will add with a comment that says only close on error.
since we do not want to close the execution state if it is to be reused.

Reply:
done